src/main/java/eu/europeana/metis/sandbox/controller/DatasetController.java

```diff
@@ -38,69 +39,116 @@
   private final DatasetReportService reportService;
+  @Value("${sandbox.dataset.max-size}")
+  private int maxRecords;
 
-  public DatasetController(ZipService zipService,
+  public DatasetController(HarvestService harvestService,
       DatasetService datasetService,
       DatasetReportService reportService) {
-    this.zipService = zipService;
+    this.harvestService = harvestService;
     this.datasetService = datasetService;
     this.reportService = reportService;
   }
 
-  @ApiOperation("Process the given dataset")
+  @ApiOperation("Process the given dataset by HTTP providing a file")
   @ApiResponses({
-          @ApiResponse(code = 202, message = MESSAGE_FOR_PROCESS_DATASET, response = Object.class)
+      @ApiResponse(code = 202, message = MESSAGE_FOR_PROCESS_DATASET, response = Object.class)
   })
-  @PostMapping(value = "dataset/{name}/process", produces = APPLICATION_JSON_VALUE)
+  @PostMapping(value = "dataset/{name}/harvestByFile", produces = APPLICATION_JSON_VALUE)
   @ResponseStatus(HttpStatus.ACCEPTED)
-  public DatasetIdDto processDataset(
+  public DatasetIdDto harvestDatasetFromFile(
       @ApiParam(value = "name of the dataset", required = true) @PathVariable(value = "name") String datasetName,
       @ApiParam(value = "country of the dataset", required = true, defaultValue = "Netherlands") @RequestParam Country country,
       @ApiParam(value = "language of the dataset", required = true, defaultValue = "nl") @RequestParam Language language,
-      @ApiParam(value = "dataset records in a zip file", required = true) @RequestParam MultipartFile dataset) {
+      @ApiParam(value = "dataset records uploaded in a zip file", required = true) @RequestParam MultipartFile dataset) {
     checkArgument(namePattern.matcher(datasetName).matches(),
         "dataset name can only include letters, numbers, _ or - characters");
-    var records = zipService.parse(dataset);
+
+    List<ByteArrayInputStream> records = harvestService.harvest(dataset);
+
     checkArgument(records.size() < maxRecords,
         "Amount of records can not be more than " + maxRecords);
 
+    var datasetObject = datasetService.createDataset(datasetName, country, language, records);
+    return new DatasetIdDto(datasetObject);
+  }
+
+  @ApiOperation("Process the given dataset by HTTP providing an URL")
+  @ApiResponses({
+      @ApiResponse(code = 202, message = MESSAGE_FOR_PROCESS_DATASET, response = Object.class)
+  })
+  @PostMapping(value = "dataset/{name}/harvestByUrl", produces = APPLICATION_JSON_VALUE)
+  @ResponseStatus(HttpStatus.ACCEPTED)
+  public DatasetIdDto harvestDatasetFromURL(
+      @ApiParam(value = "name of the dataset", required = true) @PathVariable(value = "name") String datasetName,
+      @ApiParam(value = "country of the dataset", required = true, defaultValue = "Netherlands") @RequestParam Country country,
+      @ApiParam(value = "language of the dataset", required = true, defaultValue = "nl") @RequestParam Language language,
```

Comment:
This default value here for `Language` is incorrect and needs to be updated to the latest changes done to the enum. I pointed it out as well in a different pull request [here](https://github.com/europeana/metis-sandbox/pull/31#discussion_r740116628). You can check branch MET-3501 for those changes.
Since both pull requests are very similar, I don't know which one should have this change, so I'm also writing this here as well. This applies to other calls that also contains this default value